client/src/utils/saveAsPDF.ts

```diff
@@ -0,0 +1,24 @@
+import { Election } from '@votingworks/ballot-encoder'
+
+export default async function saveAsPDF(
+  fileNamePrefix: string,
+  election: Election,
+  precinctName?: string
+): Promise<void> {
+  const precinctNameInFileName = precinctName || 'all-precincts'
+  const data = await window.kiosk!.printToPDF()
+  const fileWriter = await window.kiosk!.saveAs({
+    defaultPath: `${`${fileNamePrefix}-${election.county.name}-${election.title}`
+      .replace(/[^a-z0-9]+/gi, '-')
+      .replace(/(^-|-$)+/g, '')
+      .toLocaleLowerCase()}-${precinctNameInFileName}.pdf`,
+  })
+  if (!fileWriter) {
+    window.alert(
+      'Could not save PDF, it can only be saved to a USB device. (Or if "Cancel" was selected, ignore this message.)'
+    )
+    return
```

Comment:
This no longer belongs here since this is "UI" that is now in a utility function. I'm okay with leaving it for now, but really I think `saveAsPDF` should return a value indicating its status.